stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/MessageListItemDecoratorProvider.kt

```diff
@@ -13,15 +13,19 @@ import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.LinkAt
 import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.MaxPossibleWidthDecorator
 import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.ReactionsDecorator
 import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.ReplyDecorator
+import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.TextDecorator
+import io.getstream.chat.android.ui.messages.view.MessageListItemStyle
 
 internal class MessageListItemDecoratorProvider(
     currentUser: User,
     dateFormatter: DateFormatter,
     isDirectMessage: Boolean,
+    style: MessageListItemStyle,
```

Comment:
Isn't better to pass and keep style attributes in VHs? And provider could take those values from VHs directly

Reply:
I had it that way originally, but it seemed to me that decorators should be more concerned with keeping a style and how to apply it to each view holder, rather than having the view holder hold the style and give it to a decorator, so the decorator can then alter the view holder 🤔 What do you think?

Reply:
I think decorators only know how to do some decorations (setting background for views, applying max possible width and so on. They were introduced for avoiding code doublings for every type of VH), but ViewHolders hold/contain attributes and views for that. It's just my opinion and my way of thinking. If you see your solution as more concise and useful I don't mind anyway.